ultralytics/utils/loss.py

```diff
@@ -99,10 +99,11 @@ def __init__(self, sigmas) -> None:
     def forward(self, pred_kpts, gt_kpts, kpt_mask, area):
         """Calculates keypoint loss factor and Euclidean distance loss for predicted and actual keypoints."""
         d = (pred_kpts[..., 0] - gt_kpts[..., 0]) ** 2 + (pred_kpts[..., 1] - gt_kpts[..., 1]) ** 2
-        kpt_loss_factor = (torch.sum(kpt_mask != 0) + torch.sum(kpt_mask == 0)) / (torch.sum(kpt_mask != 0) + 1e-9)
+        kpt_mask_positive = torch.sum(kpt_mask != 0, dim=1)
+        kpt_loss_factor = (kpt_mask_positive + torch.sum(kpt_mask == 0, dim=1)) / (kpt_mask_positive + 1e-9)
```

Comment:
@yermandy isn't this line just computing the shape? i.e. if we take sum of `== 0` and sum `!= 0` that's just the total number of elements no?

Reply:
@yermandy if we want to do this for dim=1 I think maybe we could just use this equation for the numerator?

```diff
- kpt_mask_positive = torch.sum(kpt_mask != 0, dim=1)
- kpt_loss_factor = (kpt_mask_positive + torch.sum(kpt_mask == 0, dim=1)) / (kpt_mask_positive + 1e-9)
+ kpt_loss_factor = (math.prod(kpt_mask.shape) / kpt_mask.shape[1]) / ((kpt_mask != 0).sum(dim=1) + 1e-9)
```

Reply:
Or is this no longer possible since we have `.sum(1)` now instead of `.sum()`?

Reply:
Well, you are partially correct, it can be rewritten. But your solution is not correct, it should rather be:
`kpt_loss_factor =  kpt_mask.shape[1] / (torch.sum(kpt_mask != 0, dim=1) + 1e-9)`

Reply:
in other words, yes, `== 0` + `!= 0` measures the number of elements, which can be done using `kpt_mask.shape[1]` which then will be broadcasted during division

Reply:
@yermandy ah perfect! Should we implement your utils/loss.py L103?
```python
kpt_loss_factor =  kpt_mask.shape[1] / (torch.sum(kpt_mask != 0, dim=1) + 1e-9)
```

Reply:
Committed changes, thank you